Bottom: 97401cc3fd96de5c72f6412ef888183b2df9c95d
Top:    9463130200f3f1e997d58b3bcf4163093b050092
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-07 12:57:51 +0800

autofs-5.0.7 - add null check in extract_version()

A space should always be found in the passed in string but a check
should be done in case it isn't.


---

diff --git a/lib/mounts.c b/lib/mounts.c
index 0caa0aa..7b959b8 100644
--- a/lib/mounts.c
+++ b/lib/mounts.c
@@ -168,6 +168,8 @@ unsigned int get_kver_minor(void)
 static int extract_version(char *start, struct nfs_mount_vers *vers)
 {
 	char *s_ver = strchr(start, ' ');
+	if (!s_ver)
+		return 0;
 	while (*s_ver && !isdigit(*s_ver)) {
 		s_ver++;
 		if (!*s_ver)
